Libraries/Opc.Ua.Gds.Server.Common/CertificateGroup.cs

```diff
@@ -165,13 +165,18 @@ public virtual async Task Init()
             }
         }
 
-        public virtual Task<X509CRL> RevokeCertificateAsync(
+        public async virtual Task<X509CRL> RevokeCertificateAsync(
             X509Certificate2 certificate)
         {
-            return RevokeCertificateAsync(
+            var crl = RevokeCertificateAsync(
                 AuthoritiesStorePath,
                 certificate,
                 null);
+
+            //Also updated TrustedList CRL so registerd Applications get the new CRL
+            await UpdateAuthorityCertInTrustedList().ConfigureAwait(false);
```

Comment:
my understanding of the intention of this code is that, with the call to RevokeCertificateAsync, the CRL will be generated and persisted in some predefined place. The call to UpdateAuthorityCertInTrustedList() call is supposed to fetch that CRL from that place and update the CRLs in the trust store. If that perception is correct I wonder how it is ensured that new CRL was created and stored before it is fetched to be updated in the trust store. 
Would it, maybe, better to call the Update method with the help of .ContinueWith, to ensure the proper sequence of actions? Or is my perception incorrect?

Reply:
your perception is correct. I implemented ContinueWith to make the execution thread safe.
Thanks for catching this.